pkg/kube/client.go

```diff
@@ -422,6 +427,30 @@ func (c *Client) waitForJob(obj runtime.Object, name string) (bool, error) {
 	return false, nil
 }
 
+// waitForPodSuccess is a helper that waits for a pod to complete.
+//
+// This operates on an event returned from a watcher.
+func (c *Client) waitForPodSuccess(obj runtime.Object, name string) (bool, error) {
+	o, ok := obj.(*v1.Pod)
+	if !ok {
+		return true, errors.Errorf("expected %s to be a *v1.Pod, got %T", name, obj)
+	}
+
+	switch o.Status.Phase {
```

Comment:
Would it make sense to add a `default` switch case here to capture other pod statuses and report back? For example, how should the user handle pod phases like `PodUnknown`? What about other pod phases that may be introduced in the future?

Reply:
Unless we want to print a message to the user to let them know that an unknown pod status was encountered, then no. Currently if an unknown pod status is encountered then we just continue watching until `PodSucceeded` or `PodFailed` is observed.